wlp-managed-8.5/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -856,6 +860,37 @@ public void undeploy(Descriptor descriptor) throws DeploymentException {
 
    }
 
+   private void checkForDefinitionExceptions(String applicationName)
+   {
+      String messagesFile = containerConfiguration.getWlpHome() + "/usr/servers/" + containerConfiguration.getServerName() + "/logs/messages.log";
+      BufferedReader br = null;
+
+      try {
+         br = new BufferedReader(new InputStreamReader(new FileInputStream(messagesFile)));
+         String line;
+         while ((line = br.readLine()) != null) {
+            if (line.contains("CWWKZ0002E: An exception occurred while starting the application " + applicationName + ".")
+                  && (line.contains("org.jboss.weld.exceptions.DefinitionException") || line.contains("javax.enterprise.inject.spi.DefinitionException"))) {
+               System.out.println("############DEBUG found CWWKZ0002E for application: " + applicationName);
+               System.out.println(line);
+               throw new DefinitionException(line);
+            }
+         }
+      } catch (IOException e) {
+         System.err.println("Exception while reading messages.log" + e.toString());
+         e.printStackTrace();
```

Comment:
I think it might make sense here to throw a DeploymentException in this case; or do you think this can be safely ignored and just logged?

Reply:
I think you are right.